### PR TITLE
Run black over the codebase

### PR DESCRIPTION
**Pull request by corranwebster**
Updates the setup.cfg for the things that now pass.  Adds a pyproject.toml so that future runs of black have the same settings.

There is manual clean-up that is needed at this point: there are lines that were protected by `# noqa` for `E501` (line length) which no-longer make sense.  Those should be fixed in a separate PR which cleans up the `E501` status.

**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

File: ets-demo/etsdemo/app.py [hunk 1]
```diff
@@ -1021,7 +1000,7 @@ def get_children(self):
     tooltip="Go to next file",
     action="do_parent",
     enabled_when="(selected_node is not None) and "
```
Comment (rahulporuri) [inline ets-demo/etsdemo/app.py:1002]:
```suggestion
    enabled_when=(
        "(selected_node is not None) "
        "and (object.selected_node.parent is not None)"
    ),
```

File: ets-demo/etsdemo/app.py [hunk 2]
```diff
@@ -1021,7 +1000,7 @@ def get_children(self):
...
-                 "(object.selected_node.parent is not None)",
+    "(object.selected_node.parent is not None)",
```
Comment (rahulporuri) [inline ets-demo/etsdemo/app.py:1003]:
```suggestion
```

File: ets-demo/etsdemo/tests/test_app.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def get_html_tag(tag):
 
 
 def get_action_enabled(action, model_view):
-    """ Helper funciton to return if a tool is enabled.
+    """Helper funciton to return if a tool is enabled.
```
Comment (rahulporuri) [inline ets-demo/etsdemo/tests/test_app.py:39]:
```suggestion
    """Helper function to return if a tool is enabled.
```

File: ets-demo/etsdemo/tests/test_main.py [hunk 4]
```diff
@@ -35,27 +35,30 @@ def fake_configure_traits(instance):
...
-                mock.patch("sys.argv", argv):
+        with mock_iter_entry_points(
+            {}
+        ), mock_demo_launch(), mocked_io, mock.patch(
+            "sys.stdout", mocked_io
+        ), mock.patch(
+            "sys.stderr", mocked_io
+        ), mock.patch(
+            "sys.argv", argv
+        ):
```
Comment (rahulporuri) [inline ets-demo/etsdemo/tests/test_main.py:61]: sigh. these look awkward. but we can't use `(...)` in `with` statements in Python 3.6 IIUC. Not much we can do if we can to keep the codebase black-happy.

File: ets-demo/etstool.py [hunk 5]
```diff
@@ -208,30 +206,32 @@ def install(runtime, toolkit, environment, editable):
 
     # edm commands to setup the development environment
     if sys.platform == 'linux':
-        commands = ["edm environments create {environment} --platform=rh6-x86_64 --force --version={runtime}"]  # noqa: E501
+        commands = [
+            "edm environments create {environment} --platform=rh6-x86_64 --force --version={runtime}"
+        ]  # noqa: E501
```
Comment (rahulporuri) [inline ets-demo/etstool.py:211]:
```suggestion
        ]
```

File: ets-demo/etstool.py [hunk 6]
```diff
@@ -208,30 +206,32 @@ def install(runtime, toolkit, environment, editable):
...
-        commands = ["edm environments create {environment} --force --version={runtime}"]  # noqa: E501
-
-    commands.extend([
-        "edm install -y -e {environment} " + packages,
-        "edm run -e {environment} -- python setup.py clean --all",
-        install_here,
-    ])
+        commands = [
+            "edm environments create {environment} --force --version={runtime}"
+        ]  # noqa: E501
```
Comment (rahulporuri) [inline ets-demo/etstool.py:215]:
```suggestion
        ]
```

**Review by rahulporuri — APPROVED** [review]
I didnt review all of the changes black made but I trust it enough by now. LGTM. Please feel free to ignore my comments, they can be addressed separately/later.

**Comment by corranwebster** [comment]
Thanks, I deliberately avoided manual fix-ups in this PR, but the things you noticed are definitely useful for another PR which does things like sort out the `# noqa` stuff that isn't needed anymore

**Pull request merged**